drivers/tee/optee/core.c

```diff
@@ -433,38 +433,42 @@ optee_config_shm_memremap(optee_invoke_fn *invoke_fn, void **memremaped_shm,
 	size_t size;
```

Comment:
This function should be restructured. If `(sec_caps & OPTEE_SMC_SEC_CAP_DYNAMIC_SHM)` then we don't care about if there's static SHM or not since we're only going to use dynamic SHM.